### PR TITLE
[FRONT-1527] Crop image modal refactor

### PR DESCRIPTION
**Pull request by pmichalowski**

### CODE REVIEW

File: src/components/CropImageModal/CropImageModal.test.tsx [hunk 1]
```diff
@@ -28,6 +24,26 @@ jest.doMock('react-avatar-editor', () => ({
...
+            this.callOnLoad()
+        }, 0)
+    }
+
+    private async callOnLoad() {
+        if (this.onload) {
+            return this.onload()
+        }
+        await new Promise((resolve) => {
+            setTimeout(() => resolve, 10)
```
Comment (mondoreale) [inline src/components/CropImageModal/CropImageModal.test.tsx:40]:
```suggestion
            setTimeout(resolve, 10)
```

File: src/components/CropImageModal/CropImageModal.test.tsx [hunk 2]
```diff
@@ -28,6 +24,26 @@ jest.doMock('react-avatar-editor', () => ({
...
+        })
+
+        await this.callOnLoad()
```
Comment (mondoreale) [inline src/components/CropImageModal/CropImageModal.test.tsx:43]: Won't this loop forevah? If the idea is to loop-wait for `this.onload` to be set then maybe a while loop is better?

```ts
private async callOnLoad() {
    while (!this.onload) {
        await new Promise((resolve) => void setTimeout(resolve, 10));
    }

    this.onload()
}
```

File: src/components/CropImageModal/CropImageModal.test.tsx [hunk 3]
```diff
@@ -28,6 +24,26 @@ jest.doMock('react-avatar-editor', () => ({
     }),
 }))
 
+class FakeImg {
+    onload: () => void
```
Comment (mondoreale) [inline src/components/CropImageModal/CropImageModal.test.tsx:28]: Shouldn't it be

```ts
onload: undefined | (() => void)
```

?

**Review by mondoreale — COMMENTED** [review]
Hey, coming along nicely. Have a look at my comments!

File: src/components/CropImageModal/CropImageModal.test.tsx (hunk 2 above)
Comment (pmichalowski) [reply src/components/CropImageModal/CropImageModal.test.tsx:43]: It wasn't looping forever but your solution is cleaner so I'll use it :)

File: src/components/CropImageModal/CropImageModal.test.tsx (hunk 3 above)
Comment (pmichalowski) [reply src/components/CropImageModal/CropImageModal.test.tsx:28]: it's just a mock class so I don't pay so much attention to typing here :)

File: src/components/CropImageModal/CropImageModal.test.tsx (hunk 2 above)
Comment (mondoreale) [reply src/components/CropImageModal/CropImageModal.test.tsx]: Yep, second part of my comment (If blablabla) is where I realised what's all the fuss about. I know it wasn't.

**Pull request merged**